next/src/components/_global/sliderComponent/swiperComponent.tsx

```diff
@@ -0,0 +1,125 @@
+'use client';
+
+import 'swiper/css';
+import { useRef, useState } from 'react';
+import { Swiper, SwiperSlide } from 'swiper/react';
+import { A11y } from 'swiper/modules';
+import styles from './sliderComponent.module.scss';
+
+export default function SwiperComponent({ children, length }: { children: React.ReactNode[]; length: number }) {
+  const [activeIndex, setActiveIndex] = useState(0);
+
+  const swiperRef = useRef(null);
+
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-member-access
+  const handlePrev = () => swiperRef.current?.swiper?.slidePrev();
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-member-access
+  const handleNext = () => swiperRef.current?.swiper?.slideNext();
+
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access
+  const handleButtonClick = (i) => swiperRef.current?.swiper?.slideToLoop(i);
+  return (
+    <>
+      <Swiper
+        ref={swiperRef}
+        spaceBetween={50}
+        slidesPerView={1.5}
+        modules={[A11y]}
+        className={styles.slider}
+        loop={true}
+        centeredSlides={true}
+        onSlideChange={(slider) => setActiveIndex(slider.realIndex)}
+        breakpoints={{
+          0: {
+            spaceBetween: 16,
+            slidesPerView: 1.15,
+          },
+          599: {
+            spaceBetween: 16,
+            slidesPerView: 1.1,
+          },
+          1349: {
+            spaceBetween: 50,
+            slidesPerView: 1.5,
+          },
+          1600: {
+            spaceBetween: 50,
+            slidesPerView: 2.0,
+          },
+        }}
+      >
+        {Array.from({ length: length }).map((_, i) => (
+          <SwiperSlide
+            className={i == activeIndex ? `${styles.slide} ${styles.active}` : styles.slide}
+            key={i}
+          >
+            {children[i]}
+          </SwiperSlide>
+        ))}
+      </Swiper>
+      <div className={styles.controls}>
+        <ButtonLeft
+          onClick={() => {
+            handlePrev();
+          }}
+        />
+        {Array.from({ length: length }).map((_, i) => (
+          <button
+            key={i}
+            className={i == activeIndex ? `${styles.dot} ${styles.active}` : styles.dot}
```

Comment:
Don't declare the same classNames in conditions. Declare it once, or move the active indicator to the `data-*` attributes.